R/msaGaugeRR.R

```diff
@@ -532,90 +532,81 @@ msaGaugeRR <- function(jaspResults, dataset, options, ...){
+    means <- aggregate(dataset["Measurement"], dataset[parts], mean)
+
+    p <- ggplot2::ggplot()
 
-  if(options$gaugeByPartAll)
-    p <- p + jaspGraphs::geom_point(data = dataset, ggplot2::aes_string(x = parts, y = "Measurement"), col = "gray")
+    if(options$gaugeByPartAll)
+      p <- p + jaspGraphs::geom_point(data = dataset, ggplot2::aes_string(x = parts, y = "Measurement"), col = "gray")
 
+    p <- p + jaspGraphs::geom_point(data = means, ggplot2::aes_string(x = parts, y = "Measurement")) +
+      ggplot2::scale_y_continuous(limits = c(min(dataset["Measurement"]) * 0.9, max(dataset["Measurement"]) * 1.1))
 
-  p <- p + jaspGraphs::geom_point(data = means, ggplot2::aes_string(x = parts, y = "Measurement")) +
-    ggplot2::scale_y_continuous(limits = c(min(dataset["Measurement"]) * 0.9, max(dataset["Measurement"]) * 1.1))
+    p <- jaspGraphs::themeJasp(p)
 
-  p <- jaspGraphs::themeJasp(p)
+    plot$plotObject <- p
 
-  plot$plotObject <- p
+  }
 
   return(plot)
 }
 
 
-.gaugeByOperatorGraph <- function(dataset, measurements, parts, operators, options){
-
-
-  dataset <- tidyr::gather(dataset, Repetition, Measurement, measurements[1]:measurements[length(measurements)], factor_key=TRUE)
+.gaugeByOperatorGraph <- function(dataset, measurements, parts, operators, options, ready){
 
   plot <- createJaspPlot(title = gettext("Measurements by Operator"))
-
   plot$dependOn(c("gaugeByOperator", "gaugeRRmethod"))
 
-  p <- ggplot2::ggplot() +
-    ggplot2::geom_boxplot(data = dataset, ggplot2::aes_string(x = operators, y = "Measurement"))
+  if (ready){
+    dataset <- tidyr::gather(dataset, Repetition, Measurement, measurements[1]:measurements[length(measurements)], factor_key=TRUE)
 
-  p <- jaspGraphs::themeJasp(p)
+    p <- ggplot2::ggplot() +
+      ggplot2::geom_boxplot(data = dataset, ggplot2::aes_string(x = operators, y = "Measurement"))
 
-  plot$plotObject <- p
+    p <- jaspGraphs::themeJasp(p)
 
+    plot$plotObject <- p
+  }
   return(plot)
 }
 
-.gaugeByInteractionGraph <- function(dataset, measurements, parts, operators, options){
-
-  byOperator <- split.data.frame(dataset, dataset[operators])
-
-  partNames <- levels(as.factor(dataset[[parts]]))
-
-  meansPerOperator <- data.frame(Part = factor(partNames, partNames))
-
+.gaugeByInteractionGraph <- function(dataset, measurements, parts, operators, options, ready){
 
   plot <- createJaspPlot(title = gettext("Parts by Operator Interaction"), width = 700, height = 400)
-
   plot$dependOn(c("gaugeByInteraction", "gaugeRRmethod"))
 
-  for(i in 1:length(names(byOperator))){
-    name <- names(byOperator)[i]
-    if (length(rowMeans(byOperator[[name]][c(measurements)])) != length(partNames)){
-      plot$setError(gettext("Operators measured different number of parts."))
-      return(plot)
+  if (ready){
+    byOperator <- split.data.frame(dataset, dataset[operators])
+    partNames <- levels(as.factor(dataset[[parts]]))
```

Comment:
if you want to `dataset[[parts]]` to be a factor, why not read it in like that? Otherwise, this is equivalent to `unique(dataset[[parts]])`.